rs/orchestrator/src/registration.rs

```diff
@@ -158,11 +158,20 @@ impl NodeRegistration {
                         .with_identity(signer)
                         .build()
                         .expect("Failed to create IC agent");
+                    if let Err(e) = agent.fetch_root_key().await {
+                        let message = format!(
+                            "Node {} failed to cache the root key of the nns in Agent. Further responses from the IC may not be accurately decoded. Error: {}",
+                            self.node_id, e
+                        );
+                        warn!(self.log, "{}", message);
+                        UtilityCommand::notify_host(&message, 1);
+                    }
```

Comment:
As per [docs for `fetch_root_key`](https://github.com/dfinity/agent-rs/blob/main/ic-agent/src/agent/mod.rs#L265-L284) this is not advised for mainnet. I assume we can do something like 
> only do this if not on mainnet